components/MediaInfo.vue

```diff
@@ -0,0 +1,95 @@
+<template>
+  <div class="text--secondary">
+    <span v-if="item.ProductionYear && year">{{ item.ProductionYear }}</span>
+    <span v-if="item.OfficialRating && rating">{{ item.OfficialRating }}</span>
+    <span v-if="item.CommunityRating && rating"
+      ><v-icon class="rating-icon" size="16">mdi-star</v-icon>
+      {{ item.CommunityRating }}</span
+    >
+    <span v-if="item.RunTimeTicks && runtime">{{ runtimeValue }}</span>
+    <span v-if="item.RunTimeTicks && endsAt">{{
+      $t('endsAt', {
+        time: endsAtValue
+      })
+    }}</span>
+  </div>
+</template>
+
+<script lang="ts">
+import Vue from 'vue';
+import { formatDuration, intervalToDuration } from 'date-fns';
+import { BaseItemDto } from '~/api';
+import timeUtils from '~/mixins/timeUtils';
+
+export default Vue.extend({
+  mixins: [timeUtils],
+  props: {
+    item: {
+      type: Object as () => BaseItemDto,
+      required: true
+    },
+    year: {
+      type: Boolean,
+      default: false
+    },
+    rating: {
+      type: Boolean,
+      default: false
+    },
+    runtime: {
+      type: Boolean,
+      default: false
+    },
+    endsAt: {
+      type: Boolean,
+      default: false
+    }
+  },
+  computed: {
+    runtimeValue: {
+      get() {
+        const seconds = this.ticksToMs(this.item.RunTimeTicks);
+        return formatDuration(intervalToDuration({ start: 0, end: seconds }));
+      }
+    },
+    endsAtValue: {
+      get() {
+        const seconds = this.ticksToMs(this.item.RunTimeTicks);
+        return this.$dateFns.format(Date.now() + seconds, 'p');
```

Comment:
See my other comment about this mixin

Reply:
Same answer :) The mixin is to convert C# concepts into JS concepts.